org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/folding/XtextFoldingTest.java

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) 2020, 2021 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xtext.ui.editor.folding;
+
+import org.eclipse.xtext.testing.InjectWith;
+import org.eclipse.xtext.testing.XtextRunner;
+import org.eclipse.xtext.ui.testing.AbstractFoldingTest;
+import org.eclipse.xtext.xtext.ui.XtextUiInjectorProvider;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author miklossy - Initial contribution and API
+ */
+@RunWith(XtextRunner.class)
+@InjectWith(XtextUiInjectorProvider.class)
+public class XtextFoldingTest extends AbstractFoldingTest {
+	@Test
+	public void greetings() {
+		String model =
+				"grammar org.xtext.example.mydsl.MyDsl with org.eclipse.xtext.common.Terminals\n" +
+				"\n" +
```

Comment:
Check that the newline isn't an issue for windows machines!?

Reply:
no that works